libs/wingsdk/src/std/node.ts

```diff
@@ -0,0 +1,349 @@
+import {
+  IConstruct,
+  Node as ConstructsNode,
+  ConstructOrder,
+  IDependable,
+  MetadataOptions,
+  IValidation,
+} from "constructs";
+import { Connections } from "../core";
+
+const NODE_SYMBOL = Symbol.for("@winglang/sdk.std.Node");
+
+export const CONNECTIONS_FILE_PATH = "connections.json";
+export const SDK_SOURCE_MODULE = "@winglang/sdk";
+
+/**
+ * The internal node of a construct.
+ */
+export class Node {
+  /**
+   * Return the internal construct node.
+   */
+  public static of(construct: IConstruct): Node {
+    let node = (construct as any)[NODE_SYMBOL];
+
+    if (!node) {
+      node = new Node(construct);
+      (construct as any)[NODE_SYMBOL] = node;
+    }
+
+    return node;
+  }
+
+  /**
+   * Title of the construct for display purposes.
+   */
+  public title?: string;
+
+  /**
+   * Description of the construct for display purposes.
+   */
+  public description?: string;
+
+  /**
+   * The source file or library where the construct was defined.
+   */
+  public sourceModule?: string;
+
+  /**
+   * Whether the construct should be hidden by default in tree visualizations.
+   */
+  public hidden?: boolean;
+
+  private readonly _constructsNode: ConstructsNode;
+  private readonly _connections: Connections;
+
+  private constructor(construct: IConstruct) {
+    this._constructsNode = construct.node;
+    this._connections = Connections.of(construct); // tree-unique instance
+  }
+
+  /**
+   * Adds a connection between two constructs. A connection is a piece of
+   * metadata describing how one construct is related to another construct.
+   */
+  public addConnection(props: AddConnectionProps) {
+    this._connections.add(props);
+  }
+
+  // ---- constructs 10.x APIs ----
+  // https://github.com/aws/constructs/blob/10.x/src/construct.ts
+
+  /**
+   * Returns the scope in which this construct is defined.
+   *
+   * The value is `undefined` at the root of the construct scope tree.
+   */
+  public get scope(): IConstruct | undefined {
+    return this._constructsNode.scope;
+  }
```

Comment:
Pre-answering a question you might be wondering: why did I copy all of the public APIs from `constructs.Node` instead of just extending the `Node` class? The reason I did this was JSII didn't allow extending `constructs.Node` because the new signature of `std.Node.of` (returns a `std.Node`) didn't match the signature of `constructs.Node.of` (returns a `constructs.Node`). I wasn't sure if there are better alternatives 🤷

Reply:
I knew you will need to do that 😛